scripts/commands/convertApiDocsToHistorical.ts

```diff
@@ -0,0 +1,165 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { $ } from "zx";
+import { zxMain } from "../lib/zx";
+import { globby } from "globby";
+import { pathExists, getRoot } from "../lib/fs";
+import { readFile, writeFile } from "fs/promises";
+import { mkdirp } from "mkdirp";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import transformLinks from "transform-markdown-links";
+
+interface Arguments {
+  [x: string]: unknown;
+  package: string;
+}
+
+const PACKAGES: string[] = [
+  "qiskit-ibm-runtime",
+  "qiskit-ibm-provider",
+  "qiskit",
+];
+
+const readArgs = (): Arguments => {
+  return yargs(hideBin(process.argv))
+    .version(false)
+    .option("package", {
+      alias: "p",
+      type: "string",
+      choices: PACKAGES,
+      demandOption: true,
+      description: "Which package to convert",
+    })
+    .parseSync();
+};
+
+zxMain(async () => {
+  const args = readArgs();
+
+  const pkgName = PACKAGES.find((pkgName) => pkgName === args.package);
+  if (pkgName === undefined) {
+    throw new Error(`Unrecognized package: ${args.package}`);
+  }
+
+  const packageFile = await readFile(
+    `${getRoot()}/docs/api/${pkgName}/_package.json`,
+    { encoding: "utf8" },
+  );
+  const packageInfo = JSON.parse(packageFile);
+  const versionMatch = packageInfo.version.match(/^(\d+\.\d+)/);
+  const versionWithoutPatch = versionMatch[0];
+
+  const projectNewHistoricalFolder = `${getRoot()}/docs/api/${pkgName}/${versionWithoutPatch}`;
+  if (await pathExists(projectNewHistoricalFolder)) {
+    console.error(
+      `${pkgName} has already a historical version ${versionWithoutPatch}`,
+    );
+    process.exit(1);
```

Comment:
This is a good default. You're following the Principle of Least Surprise. It would be surprising if this script overwrote existing content. Instead, you make it clear to the user what's happening and ask them to decide what to do. Great job!

You could improve the error message by saying something like "Manually delete the existing folder if you intend to overwrite it."